### PR TITLE
docs: datatype map

### PR DESCRIPTION
**Pull request by soyeric128**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- added Map datatype.
- refactored Variant datatype.

Closes #issue

### CODE REVIEW

File: docs/doc/13-sql-reference/10-data-types/42-data-type-variant.md [hunk 1]
```diff
@@ -232,4 +224,4 @@ SELECT sum(arr[0]::INT) FROM array_table GROUP BY arr[0]::INT;
 
 ## Functions
 
-See [Semi-structured Functions](/doc/reference/functions/semi-structured-functions).
\ No newline at end of file
+See [Variant Functions](/doc/reference/functions/semi-structured-functions).
```
Comment (BohuTANG) [inline docs/doc/13-sql-reference/10-data-types/42-data-type-variant.md:227]: Exhaustive list of all broken links found:
--
00:23:13.115 |  
00:23:13.115 | - On source page path = /doc/sql-commands/ddl/table/ddl-create-table:
00:23:13.115 | -> linking to /doc/sql-reference/data-types/data-type-semi-structured-types
00:23:13.115 |  
00:23:13.115 | - On source page path = /doc/sql-commands/query-syntax/dml-json-path:
00:23:13.116 | -> linking to /doc/sql-reference/data-types/data-type-semi-structured-types
00:23:13.116 |  
00:23:13.116 | - On source page path = /doc/sql-reference/data-types/data-type-variant:
00:23:13.116 | -> linking to /doc/reference/functions/semi-structured-functions

**Comment by soyeric128** [comment]
> 

Thank you

**Comment by BohuTANG** [comment]
@soyeric128 

I will be continuing the work based on this pull request. 
There are several other issues that need to be addressed, including the data type index page, which requires a more technical approach :)

**Comment by BohuTANG** [comment]
Move to #10541

**Pull request closed without merging**